### DISCUSSION TITLE
Langauge dialect mappings

### DISCUSSION THREAD

**Discussion by JosephTLyons** [zed-industries/zed#6590] [Feature Requests]
I'm going to run through our langauge configuration files and add in the extensions of different dialiects of lanugages we already support, just so that Zed picks something when opening those files.  For instance, I'll add the micropython extension to the python configuration file.  Just to be clear, this will only apply an known language's highlighting to an unknown file. 

I'm going to do a bit of digging on my own, but what extensions would you like to see associated with the languages Zed already supports?

---

Batch 1 added (will ship in v0.100.x):
- C++: `cxx`, `hxx`, `inl`
- JavaScript: `cjs`
- Python: `mpy`
- TypeScript: `cts`, `d.cts`, `d.mts`, `mts`

**Comment by AmitMirgal** [comment]
Thank you, @JosephTLyons, for initiating this conversation.

Here's what I am currently seeking:

- Tailwind CSS
- Prisma ORM

**Comment by jezikk** [comment]
The following file extensions should be associated to JavaScript by default:
- .mjs
- .cjs

Example:
- prettier.config.cjs

#1822

**Comment by RemcoSmitsDev** [comment]
I am not sure if the following files could be seen as json files, because someone could use **.lock** without it needing to be json content.

1. composer.lock
2. symfony.lock

**Comment by jezikk** [comment]
Icons in project panel should match the language as well:
<img width="212" alt="image" src="https://github.com/zed-industries/community/assets/7671531/f1d254d6-5ac9-47fe-97b3-54621b658282">

**Comment by astrochili** [comment]
Hello! I would like to use Zed with the Defold game engine:

`.script` = `lua`
`.gui_script` = `lua`
`.render_script` = `lua`
`.editor_script` = `lua`

The names are a bit generic, so of course it's more desirable to be able to set the mapping by the user.

**Comment by pjjw** [comment]
technically it's not exactly the same language, but until there is actual starlark support, would be nice to map bazel's BUILD files to python. that's also not exactly extensions, but the following patterns:
```
*.bazel
*.bzl
BUILD
BUILD.*
WORKSPACE
WORKSPACE.bzlmod
```

**Comment by usizu** [comment]
not sure if it counts exactly as a dialect, but I would really love to see syntax highlighting of `.css` for `.less` files

**Comment by SomeoneToIgnore** [comment]
Now, thanks to https://github.com/zed-industries/zed/pull/9290, we have a way for users to use their local settings and set-up such dialect file associations.
